service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/OrderRemovalActions.java

```diff
@@ -21,7 +20,7 @@ public OrderRemovalAction getAction(UUID removedOrderId,
         return getActions()
             .filter(orderRemovalAction -> orderRemovalAction.isAccepted(removableOrder))
             .findFirst()
-            .orElseThrow(() -> new IllegalArgumentException(format("Action not found for order %s", removedOrderId)));
+            .orElseThrow(() -> new RemovableOrderActionNotFoundException(removedOrderId));
```

Comment:
I would say that it might be better to pass the class name than the id of the order

Reply:
or maybe pass the order and keep it with the exception and also pass the class name to the message part

Reply:
Agreed - fixed in https://github.com/hmcts/fpl-ccd-configuration/pull/1934